src/Middleware/RateLimiting/test/RateLimitingMiddlewareTests.cs

```diff
@@ -103,20 +118,218 @@ public async Task RequestRejected_WinsOverDefaultStatusCode()
     public async Task RequestAborted_ThrowsTaskCanceledException()
     {
         var options = CreateOptionsAccessor();
-        options.Value.Limiter = new TestPartitionedRateLimiter<HttpContext>(new TestRateLimiter(false));
+        options.Value.GlobalLimiter = new TestPartitionedRateLimiter<HttpContext>(new TestRateLimiter(false));
 
         var middleware = new RateLimitingMiddleware(c =>
         {
             return Task.CompletedTask;
         },
         new NullLoggerFactory().CreateLogger<RateLimitingMiddleware>(),
-        options);
+        options,
+        Mock.Of<IServiceProvider>());
 
         var context = new DefaultHttpContext();
         context.RequestAborted = new CancellationToken(true);
         await Assert.ThrowsAsync<TaskCanceledException>(() => middleware.Invoke(context)).DefaultTimeout();
     }
 
+    [Fact]
+    public async Task EndpointLimiter_Rejects()
+    {
+        var onRejectedInvoked = false;
+        var options = CreateOptionsAccessor();
+        var name = "myEndpoint";
+        options.Value.AddPolicy<string>(name, (context =>
+        {
+            return RateLimitPartition.Create<string>("myLimiter", (key =>
+            {
+                return new TestRateLimiter(false);
+            }));
+        }));
+        options.Value.OnRejected = (context, token) =>
+        {
+            onRejectedInvoked = true;
+            context.HttpContext.Response.StatusCode = 429;
+            return ValueTask.CompletedTask;
+        };
+
+        var middleware = new RateLimitingMiddleware(c =>
+        {
+            return Task.CompletedTask;
+        },
+        new NullLoggerFactory().CreateLogger<RateLimitingMiddleware>(),
+        options,
+        Mock.Of<IServiceProvider>());
+
+        var context = new DefaultHttpContext();
+        context.SetEndpoint(new Endpoint(c => Task.CompletedTask, new EndpointMetadataCollection(new RateLimiterMetadata(name)), "Test endpoint"));
+        await middleware.Invoke(context).DefaultTimeout();
+        Assert.True(onRejectedInvoked);
+        Assert.Equal(StatusCodes.Status429TooManyRequests, context.Response.StatusCode);
+    }
+
+    [Fact]
+    public async Task EndpointLimiterConvenienceMethod_Rejects()
+    {
+        var onRejectedInvoked = false;
+        var options = CreateOptionsAccessor();
+        var name = "myEndpoint";
+        options.Value.AddFixedWindowLimiter(name, new FixedWindowRateLimiterOptions(1, QueueProcessingOrder.OldestFirst, 0, TimeSpan.Zero, autoReplenishment: false));
+        options.Value.OnRejected = (context, token) =>
+        {
+            onRejectedInvoked = true;
+            context.HttpContext.Response.StatusCode = 429;
+            return ValueTask.CompletedTask;
+        };
+
+        var middleware = new RateLimitingMiddleware(c =>
+        {
+            return Task.CompletedTask;
+        },
+        new NullLoggerFactory().CreateLogger<RateLimitingMiddleware>(),
```

Comment:
nit: indenting for all the middleware blocks looks weird, since the args are aligned with the var middleware, but they are really passed in to the RateLimitingMiddleware, so feels like these should all be indented one level deeper to signify the proper nesting